pandas/io/excel.py

```diff
@@ -10,6 +10,7 @@
 import abc
 import warnings
 import numpy as np
+from http.client import HTTPResponse
```

Comment:
Did this exist in py2? May have to import from `pandas.compat`.

Reply:
Thanks for your feedback, Tom. You're right, `http.client` was named `httplib` in py2, I forgot to check py2 compatibility, sorry. So maybe `import pandas.compat.httplib as httplib` and using `httplib.HTTPResponse` would work, but I will think about your and @WillAyd's comment first.

Reply:
Shouldn't this have failed on the AppVeyor 2.7 build?

Reply:
Yes, but it worked .. I'm just trying to figure out why.

Reply:
Sorry, I've no idea why the code worked for Python 2.7 - not only in AppVeyor, but also in TravisCI and CircleCI. In all of them pytest reports to use py 2.7.14. And the [docs](https://docs.python.org/2/library/httplib.html?highlight=httplib#module-httplib) clearly state that `http.client` has come up with py3. As example, `http.client` is not available in my local py27 environment:
```
Python 2.7.14 | packaged by conda-forge | (default, Dec 25 2017, 01:16:05) 
[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from http.client import HTTPResponse
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named http.client
```
How is this possible, is there some CI magic I haven't understood yet?
I've also searched for some other compatibility libs which may have used in the pandas code base in order to achieve this, but haven't found any.

I think it would be good for me to understand this before implementing a py2 compatible fix here.

Reply:
Are you able to set up a python2 dev environment and reproduce locally? That may be easier than going through the CI logs.

Reply:
Just tested on your branch in my py2 env

```python

In [1]: import pandas as pd
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-af55e7023913> in <module>()
----> 1 import pandas as pd

/Users/taugspurger/sandbox/pandas-ip-2/pandas/pandas/__init__.pyc in <module>()
     55
     56 from pandas.util._print_versions import show_versions
---> 57 from pandas.io.api import *
     58 from pandas.util._tester import test
     59 import pandas.testing

/Users/taugspurger/sandbox/pandas-ip-2/pandas/pandas/io/api.py in <module>()
      7 from pandas.io.parsers import read_csv, read_table, read_fwf
      8 from pandas.io.clipboards import read_clipboard
----> 9 from pandas.io.excel import ExcelFile, ExcelWriter, read_excel
     10 from pandas.io.pytables import HDFStore, get_store, read_hdf
     11 from pandas.io.json import read_json

/Users/taugspurger/sandbox/pandas-ip-2/pandas/pandas/io/excel.py in <module>()
     11 import warnings
     12 import numpy as np
---> 13 from http.client import HTTPResponse
     14
     15 from pandas.core.dtypes.common import (

ImportError: No module named http.client


```

So something strange is going on.

Reply:
I wonder if one of our dependencies installs an `http` backport?

Reply:
I've tried to use `tox -e py27` in order to test with python 2.7 - that did not work, because the `bigquery` dependency could not be satisfied. I'll try to create a separate directory and build with python 2.7, that's maybe a better idea.

I'll also have a look if there are dependecies to a `http` backport...

Reply:
Hm, I've built a local development environment mit py27 and the tests fail as expected, because `http.client` cannot be imported. So it seems this is only related to the CI mechanism..

In order to check the CI dependencies, I've made a small py2 script which collects all dependencies from the files `ci/requirements-2.7(_[A-Z]+)?\.(build|pip|run)`, imports them one by one (assuming the correct version is installed) and tries to import `http.client` after each import. Haven't found a backport of `http` by doing this. 

I've added the script and my output here for reference. I had to manually install some of the collected dependencies before (couldn't find 'nomkl' btw). Not sure whether this helps.

```python

import os
import re

import logging
logging.basicConfig(level=logging.INFO)
    
if __name__=='__main__':
    DEPS_DIR = '/home/mcrot/prj/pandas-mcrot-py2/ci'
    DEPS_REGEX = r'^requirements-2.7(_[A-Z]+)?\.(build|pip|run)$'

    DEPS_BLACKLIST = ['python', 'ipython']

    DEPS_IMPORT_NAMES = { 'PyCrypto': 'Crypto',
                          'pytables': 'tables',
                          'python-dateutil': 'dateutil',
                          'beautifulsoup4': 'bs4',
                          'pandas-gbq': 'pandas_gbq'}
        
    deps = set()
    
    for fn in os.listdir(DEPS_DIR):
        if re.match(DEPS_REGEX, fn):
            logging.info("Loading deps from file '%s'..", fn)

            f = open(os.path.join(DEPS_DIR, fn))

            new_deps = []
            for s in f.readlines():
                dep = s.split('=')[0].strip()

                if dep in DEPS_BLACKLIST:
                    continue
                
                # rename if possible
                try:
                    dep = DEPS_IMPORT_NAMES[dep]
                except KeyError:
                    pass

                new_deps.append(dep)
            
            deps = deps.union(new_deps)

    deps = sorted(d for d in deps)

    logging.info("Checking these dependencies: %s", deps)
                
    for dep in deps:

        try:
            mod = __import__(dep)
        except ImportError:
            logging.warning("Cannot import module '%s'", dep)
            continue
            
        try:
            from http.client import HTTPResponse
            logging.error("Module '%s' DOES IMPORT http.", dep)
            raise Exception("Backport of http detected in module '%s'" % str(mod))
        except ImportError as exc:
            # as expected for py2
            logging.info("Module '%s' does not import http.", mod)
```

[check-backport.txt](https://github.com/pandas-dev/pandas/files/1834240/check-backport.txt)

So I guess, I need a local CI in order to find out why `http.client` can be imported?